x/observer/keeper/core_params.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k Keeper) SetCoreParamsList(ctx sdk.Context, coreParams types.CoreParamsList) {
+	store := ctx.KVStore(k.storeKey)
+	b := k.cdc.MustMarshal(&coreParams)
+	key := types.KeyPrefix(fmt.Sprintf("%s", types.AllCoreParams))
```

Comment:
Is there a reason that we can't add chain-id to the key ?
That might improve the `GetCoreParamsByChainID` slightly

Reply:
It's based on the current structure right where the params are kept as a unified list, it could be separated with a different prefix but I think it is fine right now, the performance gain would be minimal since the list of core/chain params will stay rather small depending on chain supported. Performance gain would be more if we had +10k values

Reply:
I agree the performance gain would be minimal, but I do feel. functions like add /remove or update  would look cleaner .

Reply:
I think it makes sense.
The PR is already big and necessary for v12. I created a separate issue for it: https://github.com/zeta-chain/node/issues/1541

Reply:
sounds good